src/main/java/net/ripe/rpki/rsyncit/rrdp/RrdpFetcher.java

```diff
@@ -162,11 +168,12 @@ private static void validateSnapshotStructure(int notificationSerial, String sna
         }
     }
 
-    private ProcessPublishElementResult processPublishElements(Element doc) throws XPathExpressionException {
+    private ProcessPublishElementResult processPublishElements(Element doc, Instant lastModified) throws XPathExpressionException {
         var queryPublish = XPathFactory.newDefaultInstance().newXPath().compile("/snapshot/publish");
         final NodeList publishedObjects = (NodeList) queryPublish.evaluate(doc, XPathConstants.NODESET);
 
-        var now = Instant.now();
+        var now = makeTimestampForObjects(lastModified);
+        log.debug("Will use {} as last modification timestamp for files", now);
```

Comment:
Iff we want to go this route for the limited behaviour, we need to track the modification per `(file, hash)`. Consider:
  * A snapshot contains two files of the same size that get added in the same snapshot.
  * In a newer snapshot the files get swapped

Expected behaviour: Timestamps differ.
Current behaviour: Timestamps are the same, rsync does not see the change.

Reply:
What do you mean by swapped?

Something like this

```
snapshot1.xml:
<publish uri="u1", hash="h1"/>
<publish uri="u2", hash="h2"/>

snapshot2.xml:
<publish uri="u1", hash="h2"/>
<publish uri="u2", hash="h1"/>
```
?